### ISSUE TITLE
Some TTNN operations fail with low PCC error [Grayskull and Wormhole]

### ISSUE THREAD

**Issue by banekg** [tenstorrent/tt-metal#5893]
**Describe the bug**
Some TTNN operation break with low PCC error in some test cases, for both, **Grayskull** and **Wormhole** chips. 

TTNN operations which failed are:

- ttnn.sin (passed for Wormhole)
- ttnn.sqrt
- ttnn.rsqrt

**To Reproduce**
Steps to reproduce the behavior for both chips, GS and WH:
1. Checkout `barsic/ttnn-ops` branch (to be merged into _main_ soon)
2. Run unit test `test_eltwise_sin.py ` using this command:
`pytest tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_eltwise_sin.py`

**Expected behavior**
There are test cases presented in the unit tests `pytest tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_eltwise_sin.py` and they all are expected to fail with low PCC error. 

Other unit tests (for Grayskull, and Wormhole):

```
tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_eltwise_sqrt.py
tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_eltwise_rsqrt.py
```

**Getting Additional info for the operation under test and its behavior**
To get additional information and results for different combinations of input shapes, types, layouts and memory configs for which this operation was tested you can also run locally sweeps for tt_lib.tensor.complex_abs and check the results. To do this you should:
1. Follow the `Getting Started` page to setup the repo, environment variables and `python-env`
3. Activate `source build/python_env/bin/activate`
4. Run sweeps by using `python tests/tt_eager/python_api_testing/sweep_tests/run_pytorch_test.py -i tests/ttnn/python_api_testing/sweep_tests/test_configs/ci_sweep_tests_broken/grayskull/ttnn_eltwise_sin_test.yaml -o /home/ubuntu/tt-metal/ttnn-test-sweeps/sin`
5. After the run is completed all test sweeps results should be available inside specified output directory (in this case ./result-sweeps). There you will find `eltwise_sin_sweep.csv` which holds all executed sweeps, among which you can also find the ones that failed and were recreated by the unit test, which you can get by searching unique `data_seed` field.

Other sweep tests:

```
tests/ttnn/python_api_testing/sweep_tests/test_configs/ci_sweep_tests_broken/grayskull/ttnn_eltwise_sqrt_test.yaml
tests/ttnn/python_api_testing/sweep_tests/test_configs/ci_sweep_tests_broken/grayskull/ttnn_eltwise_rsqrt_test.yaml
```

**Comment by eyonland** [comment]
None of these tests actually fail the pipeline so it's not easy to know if this is a new problem or an existing one.   I see changes as far base as three weeks ago from commits references #5137 that could have impacted the pcc.  

@banekg ,  can you provide a clarity as to whether this is a new issue where the PCC dropped?  If so, do you know what the prior PCC was on an example test?

My thought is that we might be able to use this information to do a binary search on through the list of commits to isolate which one actually introduced the new failures.

**Comment by banekg** [comment]
Hi @eyonland,

Thanks for the comment. We developed the new test sweeps which covered a large number of different input shapes for each of the operations listed above. These tests have not been run before. It is also important to mentions that the operations do not fail for all use cases, but for some specific ones.

**Comment by ruthreshx** [comment]
Hi @banekg ,
Please find the doc link from the latest main.
https://docs.google.com/document/d/1kgvpwHNC3uBAyq0ampPGPSt0NhJQoUjB9B6aj1EZdxA/edit

- Sinh & polygamma was passed on both GS & WH
- Tan supports only the range from -1.45 to 1.45
- Sin, Rsqrt and sqrt yet to debug with various range.

**Comment by nemanjagrujic** [comment]
Hello, @ruthreshx, @banekg is no longer working for TT. Can you please share doc with me (ngrujic@tenstorrent.com)?

**Comment by ruthreshx** [comment]
Hi @nemanjagrujic ,
Please verify https://docs.google.com/document/d/1kgvpwHNC3uBAyq0ampPGPSt0NhJQoUjB9B6aj1EZdxA/edit

**Comment by ruthreshx** [comment]
Hi @nemanjagrujic , @eyonland 
Can we remove the **Sinh & polygamma** was passed on both GS & WH
**Tan** supports only the range from -1.45 to 1.45,  **Sqrt & RSqrt**  supports the specific range it will be handled during the doc
updation.

Sin op not sure, I have verified the implementations of **LLK** and **Ckernel** as well. 
I will discuss with the TT team, reg the Sin issue. 
WH has passed, GS failed.

**Comment by nemanjagrujic** [comment]
@ruthreshx Removed

**Comment by ruthreshx** [comment]
> @ruthreshx Removed sink and polygamma.

Thanks @nemanjagrujic

**Comment by eyonland** [comment]
@mouliraj-mcw ,  did you discover the supported range of -30, 30 for sin on bfloat8 empirically or were you able to find this by looking at the LLKs?  Seeing "Supported input range for bfloat8_b is (-30, 30)" on your PR https://github.com/tenstorrent/tt-metal/pull/14652/files#diff-d9e29d92e93b93ec990b12569f001b1759a8442ff4f8a7e64e526d45f8459f51

**Comment by eyonland** [comment]
@rdjogoTT , I see https://docs.tenstorrent.com/pybuda/latest/dataformats.html and with BFP8-B we have a minimum positive value of 2^-127 and max positive value of 127 * 2^(255-127).  Are there underflow and overflow considerations on these and similar ops?

**Comment by mouliraj-mcw** [comment]
closing as completed.

- ttnn.tan - Handled in #6735 
- ttnn.reciprocal - Handled in #14672

**Comment by nemanjagrujic** [comment]
@mouliraj-mcw I see that unit test `tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_eltwise_tan.py` is still failing with current main.

Other unit test I see are moved or deleted so I could not test them.

**Comment by mouliraj-mcw** [comment]
Hi @nemanjagrujic 

- Regarding `tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_eltwise_tan.py ` - ttnn.tan is handled in issue https://github.com/tenstorrent/tt-metal/issues/6735.
- And reciprocal will be handled in https://github.com/tenstorrent/tt-metal/issues/14672
- Others ops has been updated in this PR https://github.com/tenstorrent/tt-metal/pull/14652

**Comment by nemanjagrujic** [comment]
> Hi [@nemanjagrujic](https://github.com/nemanjagrujic)
> 
> * Regarding `tests/ttnn/python_api_testing/non_working_unit_tests/grayskull/test_eltwise_tan.py ` - ttnn.tan is handled in issue [[Bug Report] invalid tan result  #6735](https://github.com/tenstorrent/tt-metal/issues/6735).
> * And reciprocal will be handled in [[Bug Report] ttnn.reciprocal produces incorrect results for BFLOAT8_B #14672](https://github.com/tenstorrent/tt-metal/issues/14672)
> * Others ops has been updated in this PR [#5893: Fix ops with low PCC #14652](https://github.com/tenstorrent/tt-metal/pull/14652)

ttnn.sqrt, ttnn.rsqrt and sin now pass, and tan and reciprocal are handled in other tickets so closing this.
